test/sqllogictest/comment.slt

```diff
@@ -243,20 +243,38 @@ CREATE TYPE int4_list AS LIST (ELEMENT TYPE = int4);
 statement ok
 COMMENT ON TYPE int4_list IS 'supercool_list';
 
+statement ok
+CREATE TYPE custom_type AS (x integer, y int4_list);
+
+statement ok
+COMMENT ON TYPE custom_type IS 'custom_type_comment';
```

Comment:
Should assert above this that mz_internal.mz_comments is all NULL for the inner columns.

Reply:
Added below